applications/serial_lte_modem/src/main.c

```diff
@@ -333,6 +348,39 @@ static void handle_mcuboot_swap_ret(void)
 	}
 }
 
+int lte_auto_connect(void)
+{
+	int err = 0;
+
+#if defined(CONFIG_SLM_CARRIER)
+	int stat;
+
+	if (!auto_connect) {
+		return 0;
+	}
+	err = nrf_modem_at_scanf("AT+CEREG?", "+CEREG: %d", &stat);
+	if (err != 1 || (stat == 1 || stat == 5)) {
+		return 0;
+	}
+
+	LOG_INF("auto connect");
+#if defined(CONFIG_LWM2M_CARRIER_SERVER_BINDING_N)
```

Comment:
Why the SYSTEMMODE setting has to be under CONFIG_LWM2M_CARRIER_SERVER_BINDING_N?

Reply:
> Why the SYSTEMMODE setting has to be under CONFIG_LWM2M_CARRIER_SERVER_BINDING_N?

Because modem only support NB-IoT for NIDD connection.